src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -5,106 +5,303 @@
  */
 
 #include "z_en_egol.h"
+#include "objects/object_eg/object_eg.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+#include "overlays/actors/ovl_En_Estone/z_en_estone.h"
+#include "overlays/actors/ovl_En_Arrow/z_en_arrow.h"
+#include "overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.h"
 
 #define FLAGS (ACTOR_FLAG_1 | ACTOR_FLAG_4 | ACTOR_FLAG_10 | ACTOR_FLAG_20 | ACTOR_FLAG_80000000)
 
 #define THIS ((EnEgol*)thisx)
 
+typedef enum {
+    /*  0 */ EYEGORE_ACTION_WAIT,
+    /*  1 */ EYEGORE_ACTION_STAND,
+    /*  2 */ EYEGORE_ACTION_WALK,
+    /*  3 */ EYEGORE_ACTION_RETREAT,
+    /*  4 */ EYEGORE_ACTION_LASER,
+    /*  5 */ EYEGORE_ACTION_SLAM,
+    /*  6 */ EYEGORE_ACTION_PUNCH,
+    /*  7 */ EYEGORE_ACTION_SLAM_WAIT,
+    /*  8 */ EYEGORE_ACTION_STUNNED,
+    /*  9 */ EYEGORE_ACTION_STUN_END,
+    /* 10 */ EYEGORE_ACTION_SLAM_END,
+    /* 11 */ EYEGORE_ACTION_DAMAGED,
+    /* 12 */ EYEGORE_ACTION_STOP,
+    /* 13 */ EYEGORE_ACTION_DYING,
+    /* 14 */ EYEGORE_ACTION_DEAD
+} EnEgolActionStates;
+
+typedef enum {
+    /* 0x0 */ EYEGORE_DMG_0,
+    /* 0x4 */ EYEGORE_DMG_LIGHT_ARROW = 4,
+    /* 0xE */ EYEGORE_DMG_NORMAL = 0xE,
+    /* 0xF */ EYEGORE_DMG_F,
+} EnEgolDamageEffects;
+
+typedef enum {
+    /* 0 */ EYEGORE_EFFECT_IMPACT,
+    /* 1 */ EYEGORE_EFFECT_PIECE_LARGE,
+    /* 2 */ EYEGORE_EFFECT_PIECE_SMALL,
+    /* 3 */ EYEGORE_EFFECT_DEBRIS,
+} EnEgolEffectTypes;
+
+typedef enum {
+    /*  0 */ EYEGORE_ANIM_STAND,
+    /*  1 */ EYEGORE_ANIM_WALK,
+    /*  2 */ EYEGORE_ANIM_SLAM,
+    /*  3 */ EYEGORE_ANIM_SLAM_WAIT,
+    /*  4 */ EYEGORE_ANIM_SLAM_END,
+    /*  5 */ EYEGORE_ANIM_DAMAGED,
+    /*  6 */ EYEGORE_ANIM_DEATH,
+    /*  7 */ EYEGORE_ANIM_LASER,
+    /*  8 */ EYEGORE_ANIM_LASER_END, // unused
+    /*  9 */ EYEGORE_ANIM_STUNNED,
+    /* 10 */ EYEGORE_ANIM_STUN_END,
+    /* 11 */ EYEGORE_ANIM_RETREAT,
+    /* 12 */ EYEGORE_ANIM_SIT,
+    /* 13 */ EYEGORE_ANIM_LEFT_PUNCH,
+    /* 14 */ EYEGORE_ANIM_RIGHT_PUNCH,
+    /* 15 */ EYEGORE_ANIM_MAX,
+} EnEgolAnimations;
+
+typedef enum {
+    EYEGORE_LASER_OFF,
+    EYEGORE_LASER_START,
+    EYEGORE_LASER_CHARGING,
+    EYEGORE_LASER_FIRE,
+    EYEGORE_LASER_ON = 7,
+} EnEgolLaserStates;
+
 void EnEgol_Init(Actor* thisx, PlayState* play);
 void EnEgol_Destroy(Actor* thisx, PlayState* play);
 void EnEgol_Update(Actor* thisx, PlayState* play);
 void EnEgol_Draw(Actor* thisx, PlayState* play);
 
-#if 0
-// static ColliderJntSphElementInit sJntSphElementsInit[6] = {
-static ColliderJntSphElementInit D_80A80C64[6] = {
+// Action Functions
```

Comment:
I don't feel this comment is really necessary; maybe if there were *other* forward-declared functions that *weren't* action functions, it might make sense to distinguish them with a comment. But from what I can tell, it's only action functions and the effect stuff, right?